cloud/awsprovider.go

```diff
@@ -561,3 +619,112 @@ func (*AWS) QuerySQL(query string) ([]byte, error) {
 	return nil, fmt.Errorf("Error getting query results : %s", *qrop.QueryExecution.Status.State)
 
 }
+
+type spotInfo struct {
+	Timestamp   string `csv:"Timestamp"`
+	UsageType   string `csv:"UsageType"`
+	Operation   string `csv:"Operation`
+	InstanceID  string `csv:"InstanceID`
+	MyBidID     string `csv:"MyBidID"`
+	MyMaxPrice  string `csv:"MyMaxPrice"`
+	MarketPrice string `csv:"MarketPrice"`
+	Charge      string `csv:"Charge"`
+	Version     string `csv:"Version"`
+}
+
+func parseSpotData(bucket string, prefix string, projectID string, region string, accessKeyID string, accessKeySecret string) (map[string]*spotInfo, error) {
+	log.Print("bucket " + bucket)
+	log.Print("prefix " + prefix)
+	log.Print("pID " + projectID)
+	log.Print("region " + region)
+	log.Print("accessKeyName " + accessKeyID)
+	log.Print("secret " + accessKeySecret)
+
+	if accessKeyID != "" && accessKeySecret != "" { // credentials may exist on the actual AWS node-- if so, use those. If not, override with the service key
+		os.Setenv("AWS_ACCESS_KEY_ID", accessKeyID)
+		os.Setenv("AWS_SECRET_ACCESS_KEY", accessKeySecret)
+	}
+	c := &aws.Config{
+		Region:      aws.String(region),
+		Credentials: credentials.NewEnvCredentials(),
```

Comment:
If you omit this entirely, it will do the right thing:
> Defaults to a chain of credential providers to search for credentials in environment variables, shared credential file, and EC2 Instance Roles.

https://godoc.org/github.com/aws/aws-sdk-go/aws#Config.Credentials

Which is especially important for those setups like ours that have IAM roles on the Workers to avoid requiring the hard-coding of credentials

Reply:
Thanks, that's a good point.